extensions/interactions/CodeRepl/protractor.js

```diff
@@ -0,0 +1,58 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview End-to-end testing utilities for the Pencil Code
+ * Editor.
+ */
+
+var waitFor = require('../../../core/tests/protractor_utils/waitFor.js');
+
+var customizeInteraction = function(interactionEditor, initialCode) {
+  // There are no customizations.
+};
+
+var expectInteractionDetailsToMatch = function(elem) {
+  expect(
+    elem.element(by.css('.CodeMirror')).isPresent()
+  ).toBe(true);
+};
+
+var submitAnswer = function(conversationInput, answerCode) {
+  browser.executeScript(
+    "var editor = $('.protractor-preview-tab .CodeMirror')[0].CodeMirror;" +
+    "editor.setValue('" + answerCode + "');");
```

Comment:
Right now this works because the provided code (`print("Hello World")`) uses double-quotes, but it's awfully fragile. Is there a more robust way to handle this? If not, let's put a comment by `wrongAnswers` and `correctAnswers` specifying that only double-quotes are allowed inside the string.

Reply:
It works with both single and double quotes but there has to be some changes to submit answer function.

Reply:
@rohitkatlaa I agree, but I think this fact is rather subtle. I could imagine someone later adding elements to `wrongAnswers` or `correctAnswers` with quotes that would break `submitAnswer`. For example, adding `"print('Hello')"` to `wrongAnswers` without changing `submitAnswer` would cause problems, right? This might not be clear to someone modifying this test later, which is why I think a comment would be helpful.

Reply:
I have added few comments.